source/adapters/cuda/usm.cpp

```diff
@@ -376,6 +382,48 @@ ur_result_t USMHostMemoryProvider::allocateImpl(void **ResultPtr, size_t Size,
   return USMHostAllocImpl(ResultPtr, Context, nullptr, Size, Alignment);
 }
 
+// Template helper function for creating USM pools for given pool descriptor.
+template <typename P, typename... Args>
+std::pair<ur_result_t, umf::pool_unique_handle_t>
+createUMFPoolForDesc(usm::pool_descriptor &Desc, Args &&...args) {
```

Comment:
This is pretty much the same as in L0, right? Can we implement it in common?

Reply:
Sure, I was considering moving this after both L0 and CUDA PRs with `pool_manager` are merged.